### PR TITLE
Remove dependency renaming to enable WASM

### PR DESCRIPTION
**Pull request by alubbe**
@anurbol is this what you had in mind for the follow-up PR? https://github.com/swc-project/swc/issues/106#issuecomment-581282353

### CODE REVIEW

File: ecmascript/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
-pub use ast;
-pub use codegen;
-pub use parser;
-pub use preset_env;
-pub use transforms;
+pub use swc_ecma_ast;
+pub use swc_ecma_codegen;
+pub use swc_ecma_parser;
```
Comment (kdy1) [inline ecmascript/src/lib.rs]: Please change these to ast, codegen, parser, preset_env, trnasforms, like
`pub use swc_ecma_ast as ast;`

File: ecmascript/parser/Cargo.toml [hunk 2]
```diff
@@ -42,4 +42,4 @@ pretty_assertions = "0.6"
 name = "lexer"
 
 [[example]]
-name = "typescript"
\ No newline at end of file
+name = "typescript"
```
Comment (kdy1) [inline ecmascript/parser/Cargo.toml]: If you are using windows, please set autocrlf and change line endings to lf

Comment (alubbe) [reply ecmascript/parser/Cargo.toml]: no, it's vs code on a mac

**Comment by alubbe** [comment]
Regarding the failing `fmt` test - I'm using this command, why are the outputs different from what CI would like?

```
cargo +nightly fmt -- --unstable-features --skip-children
```

File: ecmascript/transforms/Cargo.toml [hunk 3]
```diff
@@ -36,4 +36,4 @@ testing = { version = "0.5", path ="../../testing" }
 swc_ecma_codegen = { version = "0.15.0", path ="../codegen" }
 tempfile = "3"
 pretty_assertions = "0.6"
-sourcemap = "4.1.1"
\ No newline at end of file
+sourcemap = "4.1.1"
```
Comment (kdy1) [inline ecmascript/transforms/Cargo.toml]: Same as above

File: ecmascript/parser/Cargo.toml (hunk 2 above)
Comment (alubbe) [reply ecmascript/parser/Cargo.toml]: it just added a new line to a file that didn't have an empty last line

**Comment by kdy1** [comment]
> Regarding the failing `fmt` test - I'm using this command, why are the outputs different from what CI would like?
> 
> ```
> cargo +nightly fmt -- --unstable-features --skip-children
> ```

You need to add `--all` to run cargo fmt on all crates.

File: ecmascript/parser/Cargo.toml (hunk 2 above)
Comment (kdy1) [reply ecmascript/parser/Cargo.toml]: I got it.

**Comment by anurbol** [comment]
Wow, great work. As kdy already mentioned, yep it seems that you've forgotten to add aliases to the imported names (just like in javascript - you can't just rename imported variable, you should assign an alias in order for things to keep working)

**Review by kdy1 — APPROVED** [review]

**Comment by kdy1** [comment]
It seems like `cargo fmt --all` is required

**Comment by kdy1** [comment]
Btw, it's a breaking change as wasm module cannot be loaded synchronously, so transformSync and transformFileSync cannot be used.
How do you think about publishing it with a different name?

File: tests/serde.rs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 use std::path::Path;
 use swc::{
-    ecmascript::{
+    swc_ecmascript::{
```
Comment (kdy1) [inline tests/serde.rs]: This should be `ecmascript`

**Comment by alubbe** [comment]
I think `swc-wasm` should be fine

File: examples/usage.rs [hunk 5]
```diff
@@ -2,11 +2,11 @@ use swc;
 
 use std::{path::Path, sync::Arc};
 use swc::{
-    common::{
+    config::Options,
+    swc_common::{
```
Comment (kdy1) [inline examples/usage.rs]: This should be `common`

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 6]
```diff
@@ -1,9 +1,9 @@
 #![feature(box_syntax, box_patterns)]
 
-pub use atoms;
-pub use common;
-pub use ecmascript;
 pub use sourcemap;
+pub use swc_atoms;
```
Comment (kdy1) [inline src/lib.rs]: This should be `pub use swc_atoms as atoms`

**Comment by anurbol** [comment]
>wasm module cannot be loaded synchronously

@kdy1 Hi, are you sure? `new WebAssembly.Module()` is synchronous.

**Comment by kdy1** [comment]
> > wasm module cannot be loaded synchronously
> 
> @kdy1 Hi, are you sure? `new WebAssembly.Module()` is synchronous.

Oh.. i didn't know that. Then if wasm's performance is not bad, I'll change @swc/core to use wasm

**Comment by anurbol** [comment]
I don't know if you are currently making and distributing binaries, but for me that's hell. Wasm is so much easier, it's breeze. That said, wasm is 2-3 times slower than native (at least for me, on Windows), but still incredibly faster than JS (in some tasks, e.g. processing strings, x30 faster).

**Comment by anurbol** [comment]
>I've searched about it, and it seems like the size of the synchronous wasm module is limited to 4kb.  Could you share the link to that info? Seems weird. I am loading a wasm module that's a few Mb (if we talk about .wasm file size) and it works fine. 

If you do this, how are you going to sync changes and files? Is there and easy way I am not aware of?

**Comment by kdy1** [comment]
@anurbol https://developers.google.com/web/updates/2018/04/loading-wasm
Hmm... Does the limitation exist only for browser?

**Comment by anurbol** [comment]
@kdy1 I guess the limitation is only for browsers. For node.js that limitation just doesn't make sense.

**Comment by anurbol** [comment]
@kdy1 Also, I didn't even touch the WebAssembly API, I am just using [Wasm-Bindgen](https://github.com/rustwasm/wasm-bindgen) and it generates the bindings, and they are 100% synchronous. And their [book](https://rustwasm.github.io/book/) didn't mention anything about size limit.

**Comment by kdy1** [comment]
@anurbol It sounds great.

**Comment by alubbe** [comment]
It was easier for me to work all these changes into a new PR - so I'm closing this in favour of https://github.com/swc-project/swc/pull/625

**Pull request closed without merging**